### ISSUE TITLE
[Accessibility][Win11] The color contrast of the dataGridViewCheckBox cell is less than 3:1

### ISSUE THREAD

**Issue by IDEExperienceTestSolution** [dotnet/winforms#6657]
.NET Core Version: 
* .NET SDK 7.0.100-preview.2.22108.9

Have you experienced this same bug with .NET Framework?: 

- Yes

**More Info:**
1.	This issue can’t reproduce on Windows 10 OS.
2.	This issue can reproduce on Windows 11 OS both for Winforms NetFX and Core projects(can reproduce with 3.1, 5.0, 6.0, 7.0).

**Problem description:**
The color contrast less than 3:1 between background color of checkbox and background color of whole cell.
For Win11:
![image](https://user-images.githubusercontent.com/90816462/153167561-1821bee5-225b-45f9-b6aa-b4d276c29d93.png)

**Expected behavior:**
The color contrast should be greater than 3:1 between background color of checkbox and background color of whole cell.
For Win10:
![image](https://user-images.githubusercontent.com/90816462/153167703-49609c05-49e3-4643-9d7e-a619c0720e6f.png)

**Minimal repro:**
1.	Create a .NET Core Winforms application with DataGridView control added.
2.	Add a DataGridViewTextBoxColumn to it.
3.	Build and run the project.
4.	Check the checkbox, then launch Accessibility Insights for windows tool and observe the color contrast.
Or download the attached project, then execute above step 4.
[CheckBoxCellColorContrast.zip](https://github.com/dotnet/winforms/files/8031403/CheckBoxCellColorContrast.zip)

**Comment by merriemcgaw** [comment]
How about adjusting the fill color to meet the contrast ratio?

**Comment by tinodo** [comment]
You can manually override the default colors to meet the contrast requirements. But visually impaired users will typically use something like a high contrast theme in Windows. Overriding the 'default' colors with specific ones, will break this.

**Comment by NikitaSemenovAkvelon** [comment]
Style in Win11 depends on native style which you can see below on a screen of the MFC app:
![image](https://user-images.githubusercontent.com/109065597/182097409-585588ca-70e6-4f35-9acc-9bfaae7defbd.png)

I can add a black border (only for Win11) to highlight selected checkboxes. It looks like this on the screen:
![image](https://user-images.githubusercontent.com/109065597/182116418-ddcdae2a-1b77-4819-adb1-19362a062202.png)
![image](https://user-images.githubusercontent.com/109065597/182116791-ff8551a6-f066-487e-8f47-3baf106b3ecf.png)

@merriemcgaw is it an appropriate way?

**Comment by merriemcgaw** [comment]
@NikitaSemenovAkvelon I would also check to see how it looks on Win 11 in the new high contrast themes as @tinodo mentioned above. It needs to look good on all the different contrast views as well as normal view.

@Olina-Zhang I just noticed the changes in Win11 high contrast themes, it might be worth taking a look at all of our regular controls and verifying them in the new themes. I hope that some of the colors we may have hard coded don't wind up looking ridiculous. If they do look bad, we'll need to figure out a more holistic solution, so please do file an issue with the list of controls that violate the guidelines or just look plain awful 😺. That would be .NET 8 work for sure.

**Comment by NikitaSemenovAkvelon** [comment]
With any of the contrast themes, checkboxes have contrast colors. So, if a user uses a contrast theme, he won't have any trouble.
![image](https://user-images.githubusercontent.com/109065597/182767230-34af9a5d-b421-42ea-858f-1a151ec57f2b.png)
![image](https://user-images.githubusercontent.com/109065597/182767169-e845402c-5e86-4b49-8163-43108e676228.png)
![image](https://user-images.githubusercontent.com/109065597/182767115-a86700d0-3f04-40ab-9021-0ab0c612ee02.png)
![image](https://user-images.githubusercontent.com/109065597/182767057-295391c9-1c11-43ef-b3c8-7395d01a696c.png)
![image](https://user-images.githubusercontent.com/109065597/182766978-88a28f89-84c7-4420-868e-ad31e75d5e9e.png)

**Comment by merriemcgaw** [comment]
@NikitaSemenovAkvelon in that case I think you can go with your approach.

**Comment by Tanya-Solyanik** [comment]
@dmitrii-drobotov - a reliable and future-proof fix for this issue required UXTHeme team opening up some APIs for us. Putting this bug on hold until that happens,

**Comment by Tanya-Solyanik** [comment]
@merriemcgaw - should we close this issue as external?

**Comment by merriemcgaw** [comment]
I think for now we should. @KlausLoeffelmann and I are tackling getting some APIs opened on a separate track. If that opens up, we'll be able to do a lot of things.
